<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="model">
      <attribute name="title">
       <string>Model</string>
      </attribute>
      <widget class="QCheckBox" name="graph_model">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>90</y>
         <width>92</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Graph</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QCheckBox" name="matrix_model">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>40</y>
         <width>92</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Matrix</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QPushButton" name="open_button">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>200</y>
         <width>91</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Open model</string>
       </property>
      </widget>
      <widget class="QPushButton" name="new_model_button">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>240</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>New model</string>
       </property>
      </widget>
      <widget class="QPushButton" name="save_button">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>270</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>370</y>
         <width>161</width>
         <height>28</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>learning rate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="doubleSpinBox"/>
        </item>
       </layout>
      </widget>
      <widget class="QCheckBox" name="liner_func">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>310</y>
         <width>131</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Linear function</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">layer_func_group</string>
       </attribute>
      </widget>
      <widget class="QCheckBox" name="func_1">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>340</y>
         <width>191</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>1 function</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">layer_func_group</string>
       </attribute>
      </widget>
      <widget class="QCheckBox" name="func_2">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>370</y>
         <width>221</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>2 function</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">layer_func_group</string>
       </attribute>
      </widget>
      <widget class="QCheckBox" name="manual_settings">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>400</y>
         <width>161</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Manual setting</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">layer_func_group</string>
       </attribute>
      </widget>
      <widget class="QDoubleSpinBox" name="layer_k1">
       <property name="geometry">
        <rect>
         <x>630</x>
         <y>340</y>
         <width>65</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.050000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="layer_k2">
       <property name="geometry">
        <rect>
         <x>630</x>
         <y>370</y>
         <width>65</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>385</x>
         <y>11</y>
         <width>361</width>
         <height>261</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <property name="spacing">
           <number>1</number>
          </property>
          <item row="0" column="0" rowspan="2" colspan="2">
           <widget class="QGraphicsView" name="layers_view">
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAlwaysOff</enum>
            </property>
            <property name="horizontalScrollBarPolicy">
             <enum>Qt::ScrollBarAlwaysOff</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QToolButton" name="remove_layer">
                <property name="text">
                 <string>-</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="add_layer">
                <property name="text">
                 <string>+</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QLabel" name="layer_in_size">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>784</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="layers_layout"/>
          </item>
          <item>
           <widget class="QLabel" name="layer_out_size">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string> 26</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Tab 2</string>
      </attribute>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
  <buttongroup name="layer_func_group"/>
 </buttongroups>
</ui>
